db/db_impl/db_impl.cc

```diff
@@ -3122,38 +2865,29 @@ void DBImpl::PrepareMultiGetKeys(
             CompareKeyContext());
 }
 
-void DBImpl::MultiGet(const ReadOptions& read_options,
-                      ColumnFamilyHandle* column_family, const size_t num_keys,
-                      const Slice* keys, PinnableSlice* values,
-                      Status* statuses, const bool sorted_input) {
-  MultiGet(read_options, column_family, num_keys, keys, values,
-           /* timestamps */ nullptr, statuses, sorted_input);
-}
+void DB::MultiGet(const ReadOptions& _read_options,
+                  ColumnFamilyHandle* column_family, const size_t num_keys,
+                  const Slice* keys, PinnableSlice* values,
+                  std::string* timestamps, Status* statuses,
+                  const bool sorted_input) {
+  // Use an autovector, if possible, to avoid memory allocation overhead
+  autovector<ColumnFamilyHandle*, MultiGetContext::MAX_BATCH_SIZE> cf_autovec;
```

Comment:
Hmm. autovector has a vector which is essentially unused here (could have been an array). Also, constructing + destructing vector is non-trivial when not needed. There's at least 3 lines of code dedicated to not duplicating a two-line call to other MultiGet(), so I'd suggest that duplicating the call is simpler and more efficient. (Save unnecessary construction / destruction / stack allocation.)